indico/modules/events/registration/client/js/form/fields/SingleChoiceInput.jsx

```diff
@@ -11,75 +11,218 @@ import PropTypes from 'prop-types';
+            total: ((selectedChoice.extraSlotsPay ? slotsUsed : 1) * selectedChoice.price).toFixed(
+              2
+            ),
+            currency,
+          })}
+        </Label>
+      )}
+    </Form.Group>
   );
 }
 
 SingleChoiceDropdown.propTypes = {
   htmlName: PropTypes.string.isRequired,
   disabled: PropTypes.bool.isRequired,
   choices: PropTypes.arrayOf(PropTypes.shape(choiceShape)).isRequired,
-  value: PropTypes.string.isRequired,
-  onChange: PropTypes.func.isRequired,
+  isRequired: PropTypes.bool.isRequired,
+  withExtraSlots: PropTypes.bool.isRequired,
+  defaultItem: PropTypes.string,
+};
+
+SingleChoiceDropdown.defaultProps = {
+  defaultItem: null,
 };
 
-function SingleChoiceRadioGroup({htmlName, disabled, choices, value, onChange, isRequired}) {
+function SingleChoiceRadioGroup({
+  htmlName,
+  disabled,
+  choices,
+  withExtraSlots,
+  defaultItem,
+  isRequired,
+}) {
   const currency = useSelector(getCurrency);
+  const [value, setValue] = useState(defaultItem || '');
+  const [slotsUsed, setSlotsUsed] = useState(1);
+  const selectedChoice = choices.find(c => c.id === value) || {};
   const radioChoices = [...choices];
   if (!isRequired) {
     radioChoices.unshift({id: '', isEnabled: true, caption: Translate.string('None')});
   }
+
+  const handleChange = newValue => {
+    setValue(newValue);
+    setSlotsUsed(1);
+  };
+
   return (
-    <ul styleName="radio-group">
-      {radioChoices.map(c => (
-        <li key={c.id}>
-          <input
-            type="radio"
-            name={htmlName}
-            id={`${htmlName}-${c.id}`}
-            value={c.id}
-            disabled={!c.isEnabled || disabled}
-            checked={c.id === value}
-            onChange={onChange}
-          />{' '}
-          <label htmlFor={`${htmlName}-${c.id}`}>
-            {c.caption} {!!c.price && `(${c.price} ${currency})`}
-          </label>
-        </li>
-      ))}
-    </ul>
+    <table>
+      <tbody>
+        {radioChoices.map(c => {
+          return (
+            <tr key={c.id} styleName="row">
+              <td>
+                <Form.Radio
+                  styleName="radio"
+                  label={c.caption}
+                  name={htmlName}
+                  key={c.id}
+                  value={c.id}
+                  disabled={!c.isEnabled || disabled}
+                  checked={c.id === value}
+                  onChange={() => handleChange(c.id)}
+                />
+              </td>
+              <td>
+                {c.isEnabled && !!c.price && (
+                  <Label pointing="left">
+                    {c.price.toFixed(2)} {currency}
+                  </Label>
+                )}
+              </td>
+              <td>
+                {c.id !== '' && c.placesLimit !== 0 && (
+                  <PlacesLeft placesLeft={c.placesLimit} isEnabled={c.isEnabled} />
+                )}
+              </td>
+              {withExtraSlots && selectedChoice.id === c.id && (
```

Comment:
This shouldn't be shown unless the choice actually has extra slots..

![image](https://user-images.githubusercontent.com/179599/148249061-3a01fe63-db94-4263-bce9-5155a43b0ed1.png)